core/services/fluxmonitorv2/flux_monitor.go

```diff
@@ -0,0 +1,1045 @@
+package fluxmonitorv2
+
+import (
+	"fmt"
+	"math/big"
+	"reflect"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+	"github.com/shopspring/decimal"
+	"github.com/smartcontractkit/chainlink/core/assets"
+	"github.com/smartcontractkit/chainlink/core/internal/gethwrappers/generated/flags_wrapper"
+	"github.com/smartcontractkit/chainlink/core/internal/gethwrappers/generated/flux_aggregator_wrapper"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/eth"
+	"github.com/smartcontractkit/chainlink/core/services/fluxmonitor/promfm"
+	"github.com/smartcontractkit/chainlink/core/services/log"
+	"github.com/smartcontractkit/chainlink/core/services/pipeline"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+	"github.com/smartcontractkit/chainlink/core/utils"
+	"github.com/tevino/abool"
+	"go.uber.org/zap"
+	"gorm.io/gorm"
+)
+
+const hibernationPollPeriod = 24 * time.Hour
+
+// Specification defines the Flux Monitor specification
+type Specification struct {
+	ID    string
+	JobID int32
+
+	ContractAddress     common.Address
+	Precision           int32
+	Threshold           float64
+	AbsoluteThreshold   float64
+	PollTimerPeriod     time.Duration
+	PollTimerDisabled   bool
+	IdleTimerPeriod     time.Duration
+	IdleTimerDisabled   bool
+	MinJobPayment       *assets.Link
+	TransmissionAddress common.Address // The address the Eth TX is sent from
+}
+
+// FluxMonitorFactory holds the New method needed to create a new instance
+// of a FluxMonitor.
+type FluxMonitorFactory interface {
+	New(Specification, PipelineRun, Config) (*FluxMonitor, error)
+}
+
+type fluxMonitorFactory struct {
+	store          Store
+	ethClient      eth.Client
+	logBroadcaster log.Broadcaster
+}
+
+// New constructs an instance of PollingDeviationChecker with sane defaults and
+// validation.
+func (f fluxMonitorFactory) New(
```

Comment:
This factory should take the spec, create the relevant objects and inject them into the FluxMonitor with the `New` method. This allows us to more easily test separate component logic, without having to initialize a Flux Monitor each time (e.g. the sufficientPayment test)